mpf/devices/segment_display/segment_display.py

```diff
@@ -171,7 +226,7 @@ def add_text(self, text: str, priority: int = 0, key: str = None) -> None:
     def remove_text_by_key(self, key: Optional[str]):
         """Remove entry from text stack."""
         if self.config['update_method'] != "stack":
-            self.info_log("Segment display 'remove' action is TBD.")
+            self.add_text_entry("", self._previous_color, FlashingType.NO_FLASH, "", None, None, 100, key)
```

Comment:
This is a good use of an existing method to clear out a display, but the `key` part is misleading because the argument is passed to the method to define which key to _remove_, but it's actually _adding_ empty text with that key. 

I would suggest instead that this behavior check the current text to see if it has the requested key, and if so remove the text and pass an empty key, and if not do nothing.

Reply:
I would tend to disagree on this point. But maybe I misunderstood something
here. I tried to change the code as little as possible that is why we have
this early return (which I would usually try to avoid), maybe that is
misleading. Initially that function was only implemented for the `stack` case
(and that implementation has **not** changed).

For the `replace` case I would claim neither `key` nor `priority` is relevant since
we don't have a stack where we can remove something based on a key (or sort
based on a priority). Of course a `key` is passed as well for the `replace` case,
but there is nothing you can really do with it (in the `replace` case). So
removal in that case means rather emptying the display. Especially since that
remove method is as well called when the context is being removed (when a show
gets the `stop` command).

On a side note: I am working on one fix for the show `stop` since if you stop a
show then the current transition is still being played (in the `replace` case)
and a `transition_out` would be as well played. But it still empties the display.

If you really believe that the `key` is relevant for the `replace` case then please rephrase your feedback. Happy to understand it better and to improve.

Reply:
Here's a use case I'm picturing where key and priority would be helpful, let me know what you think.

A player starts a non-exclusive hurry-up mode (priority 1000) with a jackpot on the left loop, and that hurry-up mode sends the text "HIT LEFT LOOP" to a display. Instead of going for the hurry-up, the player instead starts a multi ball (priority 5000) and the multi ball mode sends the text "JACKPOT: 5M" to the display. During the multi ball, the hurry-up mode times out and ends.

By default, the hurry-up mode is set to remove the "HIT LEFT LOOP" text from the display, so it will send a command to the displays to replace the text with empty string. Without a key, the display doesn't know that the text being shown is _not_ the text that's expected to be removed, and will instead remove the "JACKPOT: 5M" text from the display even though the multi ball is still running. As a result, the multi ball will have a blank display.

Reverse the situation and start the multi ball before the hurry-up. The multi ball mode is higher priority so its display text should have precedence, and the hurry-up display text should be ignored. Otherwise the multi ball "JACKPOT: 5M" will turn into "HIT LEFT LOOP" and then disappear after the hurry-up, again leaving the multi ball with a blank display.

I realize that this type of use case is exactly what the `stack` mode is meant to do, but I believe there is value in `replace` also respecting keys and priorities because that creates more *consistent* behavior in the game. Designers will have the option to override the behavior by specifying keys and priorities if they want to, but regardless the behavior will be the same every time. Ignoring keys and priorities will yield different displays based on timing, which is not ideal.

Thoughts?

Reply:
The described use case is very valid and is very much a use case for the `stack` case. The initial implementation of `replace`

https://github.com/missionpinball/mpf/blob/7a9665173916bb6621366893c5d053d1cc329657/mpf/devices/segment_display/segment_display.py#L157

indicates that the text is written straight to the display.  Can you describe in words what would be the logic of `replace` if it includes `key` and `priority` and where this behavior differs from the `stack` case? I simply don't see the logic of such an implementation.

Reply:
You're right, I'm overthinking it and trying to keep stack-like functionality where it doesn't belong. The concept of replace doesn't need to have any considerations for priority or key or what's currently on the display, and should always overwrite when called.